backend/tardigrade/fs.go

```diff
@@ -599,6 +599,15 @@ func (f *Fs) Put(ctx context.Context, in io.Reader, src fs.ObjectInfo, options .
 
 	err = upload.Commit()
 	if err != nil {
+		if errors.Is(err, uplink.ErrBucketNotFound) {
+			// Rclone assumes the backend will create the bucket if not existing yet.
+			// Here we create the bucket and return a retry error for rclone to retry the upload.
+			_, err = f.project.EnsureBucket(ctx, bucketName)
+			if err != nil {
+				return nil, err
+			}
+			err = fserrors.RetryError(err)
```

Comment:
here `err` will be always nil. maybe you can rename error from `_, err = f.project.EnsureBucket` and with `RetryError` return error why we are retrying or make custom error like `bucket not found, create and retry`

Reply:
I passed custom error to RetryError.